docs/source/markdown/guides/how_to/data/custom_data.md

```diff
@@ -0,0 +1,289 @@
+normal and abnormal images. We could expand this example to create a
+segmentation dataset with only normal images.
+
+### With Only Normal Images
+
+There are certain cases where we only have normal images in our dataset
+but would like to train a segmentation model. Anomalib provides
+synthetic anomaly generation capabilities to create abnormal images from
+normal images. We could use the [Folder](../../reference/data/image/folder.md) datamodule to train a model on
+this dataset.
+
+:::::{dropdown} Code Syntax
+:icon: code
+
+::::{tab-set}
+:::{tab-item} API
+:sync: label-1
+
+We could run the following python code to create the custom datamodule:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/normal.txt
+:language: python
+```
+
+As can be seen from the code above, we only need to specify the
+`test_split_mode` argument to `SYNTHETIC`. The [Folder](../../reference/data/image/folder.md) datamodule will create training, validation, test and prediction datasets and
+dataloaders for us.
+
+To check what individual samples from dataloaders look like, we can run
+the following command:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/dataloader_values.txt
+:language: python
+```
+
+We could check the shape of the images and masks using the following
+commands:
+
+```python
+print(train_data["image"].shape)
+# torch.Size([32, 3, 256, 256])
+
+print(train_data["mask"].shape)
+# torch.Size([32, 256, 256])
+```
+
+:::
+
+:::{tab-item} CLI
+:sync: label-2
+
+Here is the CLI command to create the same custom datamodule with only normal images. We only need to change the `test_split_mode` argument to `SYNTHETIC` to generate synthetic anomalies.
+
+```{literalinclude} ../../../../snippets/config/data/image/folder/segmentation/cli/normal.yaml
+:language: yaml
+```
+
+Assume that we have saved the above config file as `normal.yaml`. We could run the following CLI command to train a Patchcore model on above dataset:
+
+```bash
+anomalib train --data normal.yaml --model anomalib.models.Patchcore
+```
+
+:::
+::::
+:::::
+
+## Classification Dataset
+
+Assume that we have a dataset in which the training set contains only
+normal images, and the test set contains both normal and abnormal
+images. We want to train a classification model that will be able to
+detect the abnormal images in the test set.
+
+### With Normal and Abnormal Images
+
+We could use [Folder](../../reference/data/image/folder.md) datamodule to train a model on this dataset. We
+could run the following python code to create the custom datamodule:
+
+:::::{dropdown} Code Syntax
+:icon: code
+
+::::{tab-set}
+:::{tab-item} API
+:sync: label-1
+
+```{literalinclude} ../../../../snippets/data/image/folder/classification/default.txt
+:language: python
+```
+
+```{note}
+As can be seen above, we only need to specify the ``task`` argument to ``classification``. We could have also use ``TaskType.CLASSIFICATION`` instead of ``classification``.
```

Comment:
In v1 configs we also have a `task` parameter at root level. Not sure which of these two is the intended parameter to change the task type.

Reply:
It probably sets the task in `Engine`

Reply:
Yeah, Ideally the Engine should update the task in the datasets.